### PR TITLE
Fix generation of Prometheus Rules

### PR DESCRIPTION
**Pull request by wejdross**
== Summary

```
correct one:
rate(appcat_probes_seconds_count{reason!="success", service="VSHNKeycloak",
        ha="false", maintenance="false"}[5m]) > 0.2 and rate(appcat_probes_seconds_count{reason!="success",
        service="VSHNKeycloak", ha="false", maintenance="false"}[1m]) > 0.75

this is the correct one, but on servers deployed is:
rate(appcat_probes_seconds_count{reason!="success", service="keycloak",
        ha="false", maintenance="false"}[5m]) > 0.2 and rate(appcat_probes_seconds_count{reason!="success",
        service="keycloak", ha="false", maintenance="false"}[1m]) > 0.75
```
this PR ensures that all our services ar reporting issues correctly

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/vshn_appcat_services.jsonnet [hunk 1]
```diff
@@ -196,7 +202,7 @@ local vshn_appcat_service(name, serviceParams) =
     ['22_prom_rule_sla_%s' % name]: promRuleSLA,
     [if params.services.vshn.enabled && serviceParams.enabled then 'sli_exporter/70_slo_vshn_%s' % name]: slos.Get('vshn-' + name),
     [if params.services.vshn.enabled && serviceParams.enabled then 'sli_exporter/80_slo_vshn_%s_ha' % name]: slos.Get('vshn-' + name + '-ha'),
-    [if params.slos.alertsEnabled then 'sli_exporter/90_%s_Opsgenie' % name]: opsgenieRules.GenGenericAlertingRule(name),
+    [if params.slos.alertsEnabled then 'sli_exporter/90_%s_Opsgenie' % makeVSHNServiceName(name)]: opsgenieRules.GenGenericAlertingRule(makeVSHNServiceName(name)),
```
Comment (Kidswiss) [inline component/vshn_appcat_services.jsonnet:205]: No need for another function to fix this. We already have access to the claim's kind name via the xrd:

```suggestion
    [if params.slos.alertsEnabled then 'sli_exporter/90_%s_Opsgenie' % xrd.spec.claimNames.kind]: opsgenieRules.GenGenericAlertingRule(xrd.spec.claimNames.kind),
```

File: component/vshn_appcat_services.jsonnet [hunk 2]
```diff
@@ -183,6 +183,12 @@ local vshn_appcat_service(name, serviceParams) =
     },
   };
 
+  local capitalizeFirstLetter(str) =
```
Comment (Kidswiss) [inline component/vshn_appcat_services.jsonnet:186]: By using the XRD you don't need this function anymore.

File: tests/golden/defaults/appcat/appcat/sli_exporter/90_VSHNNextcloud_Opsgenie.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  groups:
+    - name: appcat-vshnnextcloud-sla-target
+      rules:
+        - alert: vshn-vshnnextcloud-new-sla
+          annotations:
+            summary: '{{$labels.service}} {{$labels.name}} down in {{$labels.namespace}}'
+            title: '{{$labels.service}} {{$labels.name}} down in {{$labels.namespace}}'
+          expr: rate(appcat_probes_seconds_count{reason!="success", service="VSHNNextcloud",
+            ha="false", maintenance="false"}[5m]) > 0.2 and rate(appcat_probes_seconds_count{reason!="success",
+            service="VSHNNextcloud", ha="false", maintenance="false"}[1m]) > 0.75
```
Comment (wejdross) [inline tests/golden/defaults/appcat/appcat/sli_exporter/90_VSHNNextcloud_Opsgenie.yaml:20]: nextcloud sli probe doesn't exist yet, it waits for `alerting/metricts maturity level`

**Review by Kidswiss — APPROVED** [review]
LGTM

Please test that we actually get the alerts in Opgsenie now.

**Pull request merged**